x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -897,6 +899,10 @@ Collection<Object> createComponents(
 
         ipFilter.set(new IPFilter(settings, auditTrailService, clusterService.getClusterSettings(), getLicenseState()));
         components.add(ipFilter.get());
+
+        remoteClusterAuthorizationResolver.set(new RemoteClusterAuthorizationResolver(settings, clusterService.getClusterSettings()));
+        components.add(this.remoteClusterAuthorizationResolver.get());
```

Comment:
Nit: `SetOnce` is needed mostly when there is a dependency issue, i.e. core needs something from the security plugin before the thing is initialised, e.g., the `securityInterceptor`. An instance is added to `components` if it needs to be injected by guice to TransporAction classes. It seems to that that both does *not* apply to `RemoteClusterAuthorizationResolver`. In that case, I'd skip both `SetOnce` and `components` and directly use the bare `RemoteClusterAuthorizationResolver` instance, i.e. similar to how `DestructiveOperations` is handled one line below this.

Reply:
TIL. Makes sense to skip the set once part, thanks for the pointer @ywangd!